src/networking.c

```diff
@@ -3625,7 +3625,7 @@ size_t getClientOutputBufferMemoryUsage(client *c) {
         if (c->ref_repl_buf_node) {
             replBufBlock *last = listNodeValue(listLast(server.repl_buffer_blocks));
             replBufBlock *cur = listNodeValue(c->ref_repl_buf_node);
-            repl_buf_size = last->repl_offset + last->size - cur->repl_offset;
+            repl_buf_size = last->repl_offset + last->used - (cur->repl_offset+c->ref_block_pos);
```

Comment:
in fact, there's another place where if the backlog size is small, and the replication buffers are big, we attribute all the bytes beyond the configured backlog size to the replicas, but in fact, it could just be a huge node that's retained.
so that area can be problematic for very large replication buffers nodes, and will get solved as well if we limit their size.

https://github.com/redis/redis/blob/eef29b68a2cd94de1f03aa1b7891af75f5cabae2/src/evict.c#L354-L364